http3/transport.go

```diff
@@ -96,6 +97,8 @@ type Transport struct {
 	// However, if the user explicitly requested gzip it is not automatically uncompressed.
 	DisableCompression bool
 
+	Resolver *net.Resolver
```

Comment:
Do we need this? Why does the `http.Transport` not have a field like this?

Reply:
`http.Transport` does not embed a `net.Dialer`. Instead it has `Dial` callback, which can be used to inject a `net.Dialer`.

But then `http3.Transport` has such a callback as well.

Reply:
ok, I'll change it.